src/services/combineResponses.ts

```diff
@@ -272,3 +263,19 @@ function shouldCombine(frame1: DataFrame, frame2: DataFrame): boolean {
   // should never reach here
   return false;
 }
+
+function compareLabels(frame1: DataFrame, frame2: DataFrame) {
+  const field1 = frame1.fields.find((f) => f.type === FieldType.number);
+  const field2 = frame2.fields.find((f) => f.type === FieldType.number);
+  if (field1 === undefined || field2 === undefined) {
+    // should never happen
+    return false;
+  }
+  if (!frame1.name) {
+    frame1.name = JSON.stringify(field1.labels);
```

Comment:
this is a **bunch** faster so let's ship this ASAP, but I wonder if we could optimize this further by avoiding serializing the labels. Maybe by using maps and adding labels one by one from each  object and returning false as soon as we find something that doesn't match, and then generating a uuid for the name for fields that do match?

Reply:
> Maybe by using maps and adding labels one by one from each object and returning false as soon as we find something 

Unless I don't understand the suggestion, that sounds essentially equal to what we were previously doing with shallowCompare https://github.com/grafana/grafana/blob/e86283e8b173b328c4710720b8f807f359f57a77/packages/grafana-data/src/dataframe/frameComparisons.ts#L102, and even with an slightly simplified shallowCompare that I implemented for this.